src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -7402,5 +7402,15 @@ public static Tensor WrappedTensorDisposeScope(Func<Tensor> expr)
             var result = expr();
             return result.MoveToOuterDisposeScope();
         }
+        internal static Tensor InstantiateTensorWithLeakSafeTypeChange(IntPtr handle, ScalarType? dtype)
+        {
+            var tensor = new Tensor(handle);
+            if (dtype.HasValue) {
```

Comment:
Now that I see this code isolated, I'm wondering if it wouldn't be faster to check here whether the type is already the right one and skip the call to `to_type()`

Reply:
Great point I should have seen that myself, one less Tensor allocation is better. Fixed.